crates/biome_configuration/src/editorconfig.rs

```diff
@@ -121,13 +123,41 @@ impl EditorConfigOptions {
         if let Some(false) = self.insert_final_newline {
             errors.push(EditorConfigDiagnostic::incompatible(
                 "insert_final_newline",
-                "Biome always inserts a final newline.",
+                "Biome always inserts a final newline. Set this option to true.",
             ));
         }
         errors
     }
 }
 
+/// Represents a value in an .editorconfig file.
+#[derive(Debug, Clone, Deserialize, Default)]
+#[serde(untagged)]
+pub enum EditorconfigValue<T> {
+    /// The value was explicitly specified.
+    Explicit(T),
+    /// Use the default value for this option. This occurs when the value is `unset`.
+    Default,
+    /// The value was not specified.
+    #[default]
+    None,
+}
+
+// This is an `Into` because implementing `From` is not possible because you can't implement traits for a type you don't own.
+#[allow(clippy::from_over_into)]
+impl<T> Into<Option<T>> for EditorconfigValue<T>
+where
+    T: Default,
+{
```

Comment:
You can just use the inline version:
```suggestion
impl<T: Default> Into<Option<T>> for EditorconfigValue<T> {
```